OpenRA.Mods.Common/Traits/AutoTarget.cs

```diff
@@ -130,11 +133,23 @@ public void TickIdle(Actor self)
 			if (Stance < UnitStance.Defend || !Info.TargetWhenIdle)
 				return;
 
-			var allowMove = Info.AllowMovement && Stance != UnitStance.Defend;
-			if (attackFollows.All(a => a.IsTraitDisabled || !a.IsReachableTarget(a.Target, allowMove)))
+			bool allowMove;
+			if (ShouldAttack(out allowMove))
 				ScanAndAttack(self, allowMove);
 		}
 
+		bool ShouldAttack(out bool allowMove)
+		{
+			allowMove = Info.AllowMovement && Stance != UnitStance.Defend;
```

Comment:
I think so. It would need to be duplicated 3 times otherwise: inside this function, which uses it and for both callers.